types/mempool/sender_nonce.go

```diff
@@ -90,19 +90,32 @@ func SenderNonceSeedOpt(seed int64) SenderNonceOptions {
 //
 //	NewSenderNonceMempool(SenderNonceMaxTxOpt(100))
 func SenderNonceMaxTxOpt(maxTx int) SenderNonceOptions {
-	return func(snp *senderNonceMempool) {
+	return func(snp *SenderNonceMempool) {
 		snp.maxTx = maxTx
 	}
 }
 
-func (snm *senderNonceMempool) setSeed(seed int64) {
+func (snm *SenderNonceMempool) setSeed(seed int64) {
 	s1 := rand.NewSource(seed)
 	snm.rnd = rand.New(s1) //#nosec // math/rand is seeded from crypto/rand by default
 }
 
+// NextSenderTx returns the next transaction for a given sender by nonce order,
+// i.e. the next valid transaction for the sender. If no such transaction exists,
+// nil will be returned.
+func (mp *SenderNonceMempool) NextSenderTx(sender string) sdk.Tx {
+	senderIndex, ok := mp.senders[sender]
```

Comment:
The linter makes a good point, I'm somewhat distracted by differing receiver names too when reading code.